docs/html/reference/pip_show.rst

```diff
@@ -25,10 +25,49 @@ Options
 .. pip-command-options:: show
 
 
+Format
+======
+
+The various fields present in the JSON output and their explanation is as follows.
+
+*metadata*
+   A dictionary with the core metadata fields present in the ``METADATA`` file,
+   as defined in the `Core metadata specifications`_. We use the method detailed in
+   `JSON-compatible Metadata`_ to convert core metadata to json. The fields are
+   lower cased, with dashes replaced by underscores.
+
+*direct_url*
+   A dictionary containing the content of ``direct_url.json``,
+   if present, as specified in :pep:`610`.
+
+*installer*
+   A string containing the content of ``INSTALLER``,
+   if present, as specified in :pep:`376`.
+
+*record*
+   A list of ``[file_path, file_content_hash, file_size]``, representing
+   the content of ``RECORD``, if present, as specified in :pep:`376`.
+
+*requested*
+   A boolean, set to True if ``REQUESTED`` is present,
+   as specified in :pep:`376`.
+
+*required_by*
+   A list of canonicalized distribution names that depend
+   on the queried distribution.
+
```

Comment:
I would add `requires`: a list of canonicalized distribution names which this distribution depends on.

Although we'd need to think about how it would work exactly with extras.

Reply:
> I would add `requires`: a list of canonicalized distribution names which this distribution depends on.

Added in https://github.com/pypa/pip/pull/8008/commits/02759a3e87811248b6a9bc4e5db4ccdbdc6bf1b5

> Although we'd need to think about how it would work exactly with extras and environment markers.

Just to confirm, I assume that you are mentioning to [PEP-508 extras](https://www.python.org/dev/peps/pep-0508/#extras) for `extras` and [PEP-496](https://www.python.org/dev/peps/pep-0496/) for `environment markers`

It would also be helpful if you could take an example of a package, and perhaps outline what would be the expected output json in case of both, and then I can take that and the info from above links to think of a solution.

Another thing I can think of is to get out the basic version of `pip show --format=json` with what we have right now, and then think about these later. (if these are considered as advanced use cases of show by us, and can be targeted in another PR)